plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBColumnarKeyValueStorage.java

```diff
@@ -93,10 +96,33 @@ public RocksDBColumnarKeyValueStorage(
       final MetricsSystem metricsSystem,
       final RocksDBMetricsFactory rocksDBMetricsFactory)
       throws StorageException {
+    this(configuration, segments, List.of(), metricsSystem, rocksDBMetricsFactory);
+  }
+
+  public RocksDBColumnarKeyValueStorage(
+      final RocksDBConfiguration configuration,
+      final List<SegmentIdentifier> segments,
+      final List<SegmentIdentifier> ignorableSegments,
+      final MetricsSystem metricsSystem,
+      final RocksDBMetricsFactory rocksDBMetricsFactory)
+      throws StorageException {
 
     try (final ColumnFamilyOptions columnFamilyOptions = new ColumnFamilyOptions()) {
+      final List<SegmentIdentifier> trimmedSegments = new ArrayList<>(segments);
```

Comment:
Will this approach still mean it's not possible to move back to an old version of Besu e.g. 22.10.3 that doesn't have this ignore option? So we would still need to add a changelog entry having the pruning option as a breaking change for backwards compatibility?

Reply:
See above, as long as the chain pruning is not enabled, the db will be backward compatible (which is the default behaviour).